src/write/encoder_string_writer.rs

```diff
@@ -0,0 +1,107 @@
+use crate::Config;
+use std::io;
+use std::io::Write;
+use super::encoder::EncoderWriter;
+
+/// A `Write` implementation that base64-encodes data using the provided config and accumulates the
+/// resulting base64 in memory, which is then exposed as a String via `finish()`.
+///
+/// # Examples
+///
+/// ```
+/// use std::io::Write;
+///
+/// let mut enc = base64::write::EncoderStringWriter::new(base64::STANDARD);
+///
+/// enc.write_all(b"asdf").unwrap();
+///
+/// // get the resulting String
+/// let b64_string = enc.finish().unwrap();
+///
+/// assert_eq!("YXNkZg==", &b64_string);
+/// ```
+///
+/// # Panics
+///
+/// Calling `write()` (or related methods) or `finish()` after `finish()` has completed without
+/// error is invalid and will panic.
+///
+/// # Performance
+///
+/// B64-encoded data is buffered in the heap since the point is to collect it in a String.
+pub struct EncoderStringWriter {
+    encoder: EncoderWriter<Vec<u8>>,
+}
```

Comment:
It would be nice if this could write to an existing string. This would mean making it similar to the `EncoderWriter` struct taking a reference to a possibly non-empty `String`.

Reply:
Good point; no reason why not.

Reply:
I don't think this is resolved yet unless you forgot to push some code.

Reply:
I added `from()` below (open to suggestions on a better name) -- you give it a string, it gives it back to you via into_inner().